### PR TITLE
[Feat] [Ngram spec] Add triton kernel for generating tree_mask

### PR DESCRIPTION
**Pull request by SYChen123**
## Motivation
A follow up PR on #17260.

## Modifications

Write the below code into a triton kernel, to avoid the for-loop on batchsize which hinder the performance when bs is large.
image

## Accuracy Tests

gsm8k accuracy test on Qwen25-VL-72B model.

without the kernel
image

with the kernel
image

## Benchmarking and Profiling
In benchmarking, the E2E latency decreases from 5085ms to 4753ms. **332ms** performance gains per-request. 
Without the kernel
image

With the kernel
image

From the profile, we can see 
**GPU side**. When bs=32, The latency decreases from 3.875ms to **0.99ms**. 
without the kernel
image
with the kernel
image

**CPU side**. 
without the kernel
image

with the kernel
image

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

---
### CI States

Latest PR Test (Base): :x: [Run #32739557717](https://github.com/sgl-project/sglang/actions/runs/32739557717)
Latest PR Test (Extra): :x: [Run #32739557333](https://github.com/sgl-project/sglang/actions/runs/32739557333)
Latest PR Test (AMD ROCm 7.2): :x: [Run #32739558217](https://github.com/sgl-project/sglang/actions/runs/32739558217)

### CODE REVIEW

**Comment by SYChen123** [comment]
@kpham-sgl @Qiaolin-Yu @hnyls2002 Hi, This PR is ready for review. Could you please take a look when available? Thanks.

File: python/sglang/srt/speculative/ngram_worker.py [hunk 1]
```diff
@@ -299,25 +300,39 @@ def _prepare_for_speculative_decoding(self, batch: ScheduleBatch):
...
+            #         (
+            #             req_mask,
+            #             torch.from_numpy(mask[i]).to(
+            #                 device=self.device, non_blocking=True
+            #             ),
+            #         ),
+            #         dim=1,
+            #     ).to(torch.bool)
+            #     tree_mask.append(req_mask.flatten())
+            # tree_mask = torch.cat(tree_mask, dim=0)
```
Comment (kpham-sgl) [inline python/sglang/srt/speculative/ngram_worker.py]: You can remove this comment

File: python/sglang/srt/speculative/spec_utils.py [hunk 2]
```diff
@@ -628,3 +630,76 @@ def commit_mamba_states_after_verify(
             mamba_steps_to_track=mamba_steps_to_track,
             model=model_runner.model,
         )
+
+
+@triton.jit
+def generate_tree_mask(
```
Comment (kpham-sgl) [inline python/sglang/srt/speculative/spec_utils.py:1035]: Need an unit test for this kernel

**Comment by SYChen123** [comment]
@kpham-sgl Done. Thanks for your review. I've cleaned the comment and add the ut test, test_generate_tree_mask.py. The test covers different combs of seq_len and draft_token_num. 

image
